service/service.go

```diff
@@ -301,14 +333,15 @@ func (s *Service) Recover() error {
 				Add(s.cfg.PhaseShift).
 				Add(-s.cfg.CycleGap)
 
-			if err = r.recoverExecution(rs.Execution, end); err != nil {
+			if err = r.recoverExecution(ctx, rs.Execution, end); err != nil {
 				s.asyncError(fmt.Errorf("recovery: round %v execution failure: %v", r.ID, err))
-				return
+				return err
```

Comment:
Is `asyncError` now even still required? Or could we just change that to:

```suggestion
			if err := r.recoverExecution(ctx, rs.Execution, end); err != nil {
				err = fmt.Errorf("recovery: round %v execution failure: %w", r.ID, err)
				log.Error(err.Error())
				return err
			}
```

Is `s.errChan` still in use?

Reply:
Unfortunately, tests depend on it.

Reply:
- [ ] `Service::errChan` is never read so failed rounds will block